Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -962,13 +966,47 @@ public extension Purchases {
 
     @objc(purchasePackage:withCompletion:)
     func purchase(package: Package, completion: @escaping PurchaseCompletedBlock) {
-        purchasesOrchestrator.purchase(product: package.storeProduct, package: package, completion: completion)
+        purchasesOrchestrator.purchase(product: package.storeProduct,
+                                       package: package,
+                                       promotionalOffer: nil,
+                                       metadata: nil,
+                                       completion: completion)
     }
 
     func purchase(package: Package) async throws -> PurchaseResultData {
         return try await purchaseAsync(package: package)
     }
 
+    #if ENABLE_PURCHASE_PARAMS
+
+    @objc(purchaseProduct:params:withCompletion:)
+    func purchase(product: StoreProduct, params: PurchaseParams, completion: @escaping PurchaseCompletedBlock) {
```

Comment:
I was wondering, should the `StoreProduct`/`Package` be part of the `PurchaseParams`? We can create multiple `Builder` constructors, one for each of these, with the mandatory parameters. FWIW, this is how it works in Android.

Reply:
I like the idea of simplifying the purchase methods and including the `StoreProduct/Package` be part of the `PurchaseParams`. However I don't think there are any mandatory parameters beyond the product/package itself. All the rest are optional.

Reply:
Note that maybe we will need a third once we support win back offers? cc @fire-at-will. And mostly this allows us to have a single purchase entry point and the rest being implementation details inside the `PurchaseParams`. If there is a reason to want to keep different purchase methods I would be ok, but otherwise I feel that's a cleaner API IMO.

Reply:
I added an initialiser with product and another with package to `PurchaseParams.Builder` and removed it from the purchase method. Makes it much cleaner and can help us add better validation inside `PurchaseParams` 👍